### ISSUE TITLE
onChild* events not triggered on touch devices

### ISSUE THREAD

**Issue by deguilardi** [google-map-react/google-map-react#959]
On touch devices (tested on androids and raspberry pis), onChild* events aren't called well.

mouseDown and mouseUp are called on touch devices both together at the moment the touch is finished, when you take your finger off of the screen. This behavior makes the onChildMouse* events pointless.

````
<GoogleMapReact
          onChildMouseDown={(a,b,c) => {console.log('down',a,b,c)}}
          onChildMouseUp={(a,b,c) => {console.log('up',a,b,c)}}>
          onChildMouseMove={(a,b,c) => {console.log('move',a,b,c)}}
    ...
</GoogleMapReact>
````

To reproduce this:
1. open the app on chrome
2. then activate "device bar" and pick any touch device from the list
3. hit a child (marker)

The following console.log do work with mouse events. When on a touch device (or chrome simulator), mouseMove is never called.

Super easy reproducible live example:
1. access https://jsbin.com/molerasebo/2/edit?js,console,output
2. add `draggable={false}` to the <GoogleMapReact> props
3. activate "device simulator" or access from your tablet
You will see the red marker won't be draggable anymore

**Comment by deguilardi** [comment]
I know why making touch to work is so difficult.
They base the whole mouse actions on a hover logic.
To know if the touch / click is on top of a marker, the architecture depends on knowing where the cursor is.
This just won't work with touch, because touch devices don't have a cursor

I spent a few hours to fix this.
I had to rewrite the hover thing. And as my use case is for touch devices only, I created a type of "touch/click" switch.
So, if it's set to work with touch devices, it won't work with mouse devices, and vice-versa.

I won't push this solution because I don't see it as a all-case general thing.
Let me know if anyone wants it.

**Comment by parthpatelsolulab** [comment]
Hi @deguilardi ,

I would like to know this solution please provide it to me. Thanks.

**Comment by deguilardi** [comment]
@parthpatelsolulab https://github.com/carebnb-app/google-map-react

**Comment by parthpatelsolulab** [comment]
> @parthpatelsolulab https://github.com/carebnb-app/google-map-react

Thanks @deguilardi,

I tried this package and double check this the changes comes in my local but still I am having the same issue.

**Comment by deguilardi** [comment]
@parthpatelsolulab. package? What do you mean? Read carefully, I didn't even made a pull request. It will never be in any package if it's not merged.

**Comment by dowrow** [comment]
@deguilardi I believe @parthpatelsolulab clearly referred to your repo, which didn't solve the issue at hand.

**Comment by deguilardi** [comment]
Well, it works with me. Tested on ipad, raspberry pi,l and android.

To make it work in a computer, you must activate the device simulation
mode, as tap events aren't triggered by a mouse.

**Comment by dowrow** [comment]
Sorry @deguilardi, to be clear, I didn't manage to make your code work. 
I replaced my dependency by the index of your project and it didn't load the Google library properly, so I don't know if this really fixes the issue or not. That being said, I don't see why you wouldn't want to transform this into a legit PR, that would at least enforce interface compliance.

**Comment by itsmichaeldiego** [comment]
Yes @deguilardi , please submit a PR, we would love to have this!

**Comment by Bharateshwar** [comment]
Hi @itsmichaeldiego .. do we have a stable fix for this issue? I really loved this lib.. and kind of developed a whole lot over it. But not being
